sheets/api.py

```diff
@@ -570,13 +631,12 @@ def parse_row_and_create_coupons(row_index, row_data):
                 and the data in the sheet do not agree
             SheetCouponCreationException: Raised if there was an error during coupon creation
         """
-        purchase_order_id = row_data[CouponRequestRow.PURCHASE_ORDER_COL_INDEX].strip()
+        coupon_name = row_data[CouponRequestRow.COUPON_NAME_COL_INDEX].strip()
         user_input_json = json.dumps(CouponRequestRow.get_user_input_columns(row_data))
 
         with transaction.atomic():
             coupon_gen_request, created = CouponGenerationRequest.objects.select_for_update().get_or_create(
-                purchase_order_id=purchase_order_id,
-                defaults=dict(raw_data=user_input_json),
+                coupon_name=coupon_name, defaults=dict(raw_data=user_input_json)
```

Comment:
Should `purchase_order_id` still be passed in defaults so it isn't blank in the `CouponGenerationRequest` model?